### PR TITLE
More control flow syntax

### PR DESCRIPTION
**Pull request by stephanh**
Added some more functionality to LoadInfo and MaestroExecution to assist with improving control flow code.

### CODE REVIEW

File: maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/task/LoadExecution.scala [hunk 1]
```diff
@@ -45,27 +45,40 @@ import au.com.cba.omnia.omnitool.Result
...
     * Fails if we should not continue
     */
-  def withSuccess: Execution[LoadSuccess] = Execution.from(this match {
-    case ls@LoadSuccess(_, _, _, _) => ls
-    case _                          => throw new Exception(s"Unsuccessfull load. Load info = $this")
-  })
+  def withSuccess: Execution[LoadSuccess] = fold(
+    Execution.from(_),
+    Execution.fail(s"Emtpy load"),
+    lf => Execution.fail(s"Unsuccessfull load. Load info = $lf")
```
Comment (stephanh) [inline maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/task/LoadExecution.scala]: Codacy Issue found: [The string literal " appears 7 times in the file.](https://www.codacy.com/app/CommBank/maestro/file/485526623/issues/source?bid=2205397&fileBranchId=2253054#l77)

File: maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/scalding/MaestroExecution.scala [hunk 2]
```diff
@@ -31,4 +31,7 @@ object MaestroExecution {
       case JobNotReadyException          => Execution.from(JobNotReady)
       case JobFailureException(exitCode) => Execution.from(JobFailure(exitCode))
     }
+
+  /** Iff the condition is not true changes the status of the Execution to JobNotReady. */
+  def guardReady(check: Boolean): Execution[Unit] = if (check) Execution.from(()) else jobNotReady
```
Comment (SamRoberts) [inline maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/scalding/MaestroExecution.scala]: Maybe a monadic version of this method too?

**Comment by SamRoberts** [comment]
:+1:

File: maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/scalding/MaestroExecution.scala (hunk 2 above)
Comment (stephanh) [inline maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/scalding/MaestroExecution.scala]: I wasn't going to add one just yet since this is mostly going to be used inside a for comprehension.

Comment (stephanh) [inline maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/scalding/MaestroExecution.scala]: Happy to be convinced otherwise and do another PR.

Comment (SamRoberts) [inline maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/scalding/MaestroExecution.scala]:
```
exists <- checkExists
guardReady(exists)
```

vs.

```
guardReady(checkExists)
```

Up to you :smile:
